apps/web/src/utils/graphql/experiences/useUpdateUserExperience.tsx

```diff
@@ -5,14 +5,22 @@ import { useToastActions } from '~/contexts/toast/ToastContext';
 import {
   UserExperienceType,
   useUpdateUserExperienceMutation,
+  useUpdateUserExperienceMutation$data,
 } from '~/generated/useUpdateUserExperienceMutation.graphql';
 import { usePromisifiedMutation } from '~/hooks/usePromisifiedMutation';
 
 type Props = {
   type: UserExperienceType;
   experienced: boolean;
+  optimisticExperiencesList: ReadonlyArray<{
+    readonly experienced: boolean;
+    readonly type: UserExperienceType;
+  }> | null;
```

Comment:
@tbezman i couldn't figure out how to avoid doing this manually given the response type could vary between a valid payload or `ErrInvalidInput`

Reply:
This is totally fine